### ISSUE TITLE
Custom Plugins

### ISSUE THREAD

**Issue by aksk1119** [swri-robotics/mapviz#769]
Hi all,
Is there a way to add a custom plugin to Mapviz?

Thank you

**Comment by malban** [comment]
Yes,  you can look at the [mapviz_plugins](https://github.com/swri-robotics/mapviz/tree/master/mapviz_plugins), [multires_image](https://github.com/swri-robotics/mapviz/tree/master/multires_image), and [tile_map](https://github.com/swri-robotics/mapviz/tree/master/tile_map) packages as examples.

You'll need a [mapviz_plugins.xml](https://github.com/swri-robotics/mapviz/blob/master/mapviz_plugins/mapviz_plugins.xml) file and to reference it in [package.xml](https://github.com/swri-robotics/mapviz/blob/master/mapviz_plugins/package.xml#L49) so mapviz can find the plugin(s).

Basically, you will inherit from the `mapviz::MapvizPlugin` base class and implement the custom functionality.  For example: https://github.com/swri-robotics/mapviz/blob/master/mapviz_plugins/include/mapviz_plugins/grid_plugin.h

**Comment by aksk1119** [comment]
Thank you, Marc. I will take a look at them!
